Doc/library/importlib.rst

```diff
@@ -420,30 +420,34 @@ ABC hierarchy::
         reloaded):
 
         - :attr:`__name__`
-            The name of the module.
+            The module's fully-qualified name.
 
         - :attr:`__file__`
-            The path to where the module data is stored (not set for built-in
-            modules).
+            The location of the module's source code.
+            It is ``None`` for namespace packages and not set for built-in
+            modules and frozen modules.
 
         - :attr:`__cached__`
-            The path to where a compiled version of the module is/should be
-            stored (not set when the attribute would be inappropriate).
+            The location of the module's compiled code.
+            It is not set for built-in modules, frozen modules
+            and namespace packages.
 
         - :attr:`__path__`
-            A list of strings specifying the search path within a
-            package. This attribute is not set on modules.
+            The iterable (possibly empty) of the locations of the module's
```

Comment:
I agree with your suggestion. We could also add "search path" in parentheses in the second sentence so that the term does not go away:

> The iterable (possibly empty) of all locations to search (search path) when looking for the package's submodules.

Reply:
re-visited in a follow-up review